data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetBlockRoot.java

```diff
@@ -73,16 +101,22 @@ public GetBlockRoot(final ChainDataProvider chainDataProvider, final JsonProvide
       })
   @Override
   public void handle(@NotNull final Context ctx) throws Exception {
-    final Map<String, String> pathParams = ctx.pathParamMap();
-    final SafeFuture<Optional<ObjectAndMetaData<Root>>> future =
-        chainDataProvider.getBlockRoot(pathParams.get(PARAM_BLOCK_ID));
-    handleOptionalResult(ctx, future, this::handleResult, SC_NOT_FOUND);
+    adapt(ctx);
   }
 
-  private Optional<String> handleResult(Context ctx, final ObjectAndMetaData<Root> response)
-      throws JsonProcessingException {
-    return Optional.of(
-        jsonProvider.objectToJSON(
-            new GetBlockRootResponse(response.getData(), response.isExecutionOptimisticForApi())));
+  @Override
+  public void handleRequest(RestApiRequest request) throws JsonProcessingException {
+    final SafeFuture<Optional<ObjectAndMetaData<Bytes32>>> future =
+        chainDataProvider.getBlockRoot(request.getPathParameter(PARAMETER_BLOCK_ID));
```

Comment:
we could follow the same pattern of getting the block and then using the call to get the root here if we wanted... would fit into a generic handler then... just a thought...

Reply:
There's only 3 I guess so it might not be worth it but they're all v. similar in that they just pull data off the block

Reply:
Yes might not be worth it yet, but will keep it in mind for the future